cmd/server.go

```diff
@@ -80,27 +80,44 @@ func serverMain(cmd *cobra.Command, args []string) exitCode {
 		}()
 	}
 
+	// On Ctrl-C, trigger the clean-up. This consists of shutting down the API
+	// server and unmounting the FS.
+	sigCh := make(chan os.Signal, 1)
+	signal.Notify(sigCh, syscall.SIGINT, syscall.SIGTERM)
+
+	opts := serverOpts{cpuprofile: cpuprofile, externalPluginsPath: externalPluginsPath}
+	if err := runServer(mountpoint, make(chan struct{}), sigCh, opts); err != nil {
+		log.Warn(err)
+		return exitCode{1}
+	}
+	return exitCode{0}
+}
+
+type serverOpts struct {
+	cpuprofile, externalPluginsPath string
+}
+
+func runServer(mountpoint string, readyCh chan struct{}, sigCh chan os.Signal, opts serverOpts) error {
```

Comment:
This indicates that it might be a good time to introduce a `plugin.Server` object to encapsulate some of this code. Then you can have something like `server.Start() readyCh chan error`, `server.Stop() stoppedCh chan error`. Thinking `plugin.Server` could replace `plugin.Registry`, so you can also do `server.RegisterCorePlugin()`, `server.RegisterExternalPlugin()`, both of which read well. These changes would also make it easier for third parties to use the Wash server if they need to.

Reply:
I don't think it should be part of plugin (for one, that introduces a circular dependency between fuse and plugin), but an object seems to make sense.

Reply:
Agreed. I think `api/server.go` could contain the object, and we could move the FUSE package over to an internal package in the API since I don't think that's something we want people to use.

`api.NewServer()` also reads well.

Reply:
I started this with in an independent internal package for now to give us a little more flexibility. I'm not sure it's generally useful yet.